src/main/java/duke/TaskList.java

```diff
@@ -0,0 +1,169 @@
+package duke;
+
+import java.io.FileNotFoundException;
+import java.util.ArrayList;
+
+import duke.exception.DukeException;
+import duke.exception.DukeInvalidDateException;
+import duke.exception.DukeInvalidDayException;
+import duke.exception.DukeInvalidTaskException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+/**
+ * The TaskList class is used to keep track of the tasks given by the user.
+ * Interacts with a storage object when changes to the task list is made.
+ */
+
+public class TaskList {
+    private ArrayList<Task> todo;
+    private Storage storage;
+
+    /**
+     * Initalizes a TaskList object
+     *
+     * @param storage object in which the file data is read, written and stored in.
+     * @throws FileNotFoundException
+     * @throws DukeException
+     */
+    public TaskList(Storage storage) throws FileNotFoundException, DukeException {
+        this.storage = storage;
+        this.todo = storage.load();
+    }
+
+
+    /**
+     * Deletes the task from the task list.
+     *
+     * @param line the line in which the command of delete was given.
+     * @throws DukeException
+     */
+
+    public void delete(String line) throws DukeException {
+        String[] splits = line.split("delete ");
+        try {
+            if (splits.length > 1) {
+                int taskNumber = Integer.parseInt(splits[1]);
+                if (taskNumber - 1 >= todo.size() || taskNumber - 1 < 0 || todo.size() == 0) {
+                    throw new DukeException("Invalid task number");
+                } else {
+                    int size = todo.size() - 1;
+                    Ui.showLine();
+                    System.out.println("Noted. I've removed this task: \n"
+                            + todo.get(taskNumber - 1) + "\n" + "Now you have " + size + " task in the list \n"
```

Comment:
This is slightly over 110 characters. Since it's already being wrapped from Line 54 maybe you could wrap it again to another line?